addon/components/password-toggle.js

```diff
@@ -2,20 +2,17 @@ import Ember from 'ember';
 
 export default Ember.Component.extend({
     didInsertElement: function() {
-        $('button').on('click', function(){
-          var text = $(this).text();
-          var $parent = $(this).parent();
-          var $input = $('input', $parent).clone();
+        $('button').on('click', function() {
+            var text = $(this).text();
+            var $input = $(this).parent().find('input');
 
-          $('input', $parent).remove();
```

Comment:
I did FF and Chrome (they work fine), and was poised to try IE9 but didn't
want to stay at work long enough to reboot my laptop so the VirtualBox
driver could load after installing.

If this is a no-go because of IE or whatever, I think we'd have to migrate
the event listeners over after the clone, probably. I don't know how to do
that, but must be possible. :-P
On Jan 27, 2015 7:57 PM, "Toran Billups" <notifications@github.com> wrote:

> In addon/components/password-toggle.js
> <https://github.com/nchristus/ember-cli-password-toggle/pull/5#discussion_r23660733>
> :
>
> >
> > -          $('input', $parent).remove();
>
> I remember talking with Nick about why we decided to clone instead of
> hide/show and it had something to do with browser support. Nick -can you
> confirm this non clone version (the one in this PR) will work on IE9/10,
> Firefox, Chrome, iOS 8 Safari before we merge it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nchristus/ember-cli-password-toggle/pull/5/files#r23660733>
> .
>